File: scratch/fix_sql.py
with open(r"d:\Ashik\Sheba SQL backup\shebafi_minhaj.sql", "r", encoding="utf-8", errors="ignore") as f:
    sql = f.read()

# Replace unknown collations
sql = sql.replace("utf8mb4_0900_ai_ci", "utf8mb4_unicode_ci")

with open(r"d:\Ashik\Shebad 21 may\scratch\shebafi_minhaj_fixed.sql", "w", encoding="utf-8") as f:
    f.write(sql)

print("SQL collation fixed successfully.")

File: scratch/fix_sql.php
<?php
$inputFile = 'd:\\Ashik\\Sheba SQL backup\\shebafi_minhaj.sql';
$outputFile = 'd:\\Ashik\\Shebad 21 may\\scratch\\shebafi_minhaj_fixed.sql';

if (!file_exists($inputFile)) {
    die("Input file not found: $inputFile\n");
}

$sql = file_get_contents($inputFile);
if ($sql === false) {
    die("Failed to read input file.\n");
}

// Replace collation
$sql = str_replace('utf8mb4_0900_ai_ci', 'utf8mb4_unicode_ci', $sql);

if (file_put_contents($outputFile, $sql) === false) {
    die("Failed to write output file.\n");
}

echo "SQL collation fixed successfully.\n";
?>
